opm/autodiff/NewtonIterationBlackoilInterleaved.cpp

```diff
@@ -120,23 +182,39 @@ namespace Opm
             typedef std::unique_ptr<typename ScalarProductChooser::ScalarProduct> SPPointer;
             SPPointer sp(ScalarProductChooser::construct(parallelInformation_arg));
 
-            // Construct preconditioner.
-            auto precond = constructPrecond(opA, parallelInformation_arg);
-
             // Communicate if parallel.
             parallelInformation_arg.copyOwnerToAll(istlb, istlb);
 
-            // Solve.
-            solve(opA, x, istlb, *sp, *precond, result);
-        }
+#if ! HAVE_UMFPACK
+            const bool useAmg = false ;
+            if( useAmg )
+            {
+                typedef ISTLUtility::CPRSelector< Mat, Vector, Vector, POrComm>  CPRSelectorType;
+                typedef typename CPRSelectorType::AMG AMG;
+                std::unique_ptr< AMG > amg;
+                // Construct preconditioner.
+                constructAMGPrecond(opA, parallelInformation_arg, amg);
+
+                // Solve.
+                solve(opA, x, istlb, *sp, *amg, result);
+            }
+            else
+#endif
+            {
+                // Construct preconditioner.
+                auto precond = constructPrecond(opA, parallelInformation_arg);
 
+                // Solve.
+                solve(opA, x, istlb, *sp, *precond, result);
+            }
+        }
 
         typedef Dune::SeqILU0<Mat, Vector, Vector> SeqPreconditioner;
 
         template <class Operator>
         std::unique_ptr<SeqPreconditioner> constructPrecond(Operator& opA, const Dune::Amg::SequentialInformation&) const
         {
-            const double relax = 1.0;
+            const double relax = 0.9;
```

Comment:
There is two methods contructPrecond because on is specialized for non-parallel runs. 
Once we stop supporting flow without MPI we can drop one method.  I think the relax factor should be a parameter anyway, but it was you who coded it like this in the first place, even when we have introduced a parameter relaxation factor before.